File: VB-Examples/30_Macros/Macros.vb
Imports System.IO
Imports System.Xml.XPath
Imports Spire.Doc
Imports Spire.Doc.Documents
Imports Spire.Doc.Fields

Namespace Macros
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Create a new instance of Document
			Dim document As New Document()

			' Load the Word document from the specified file that may contain VBA macros
			document.LoadFromFile("../../../../../../Data/Macros.docm", FileFormat.Docm)

			' Save the document to a new file with the specified name and format (Docm for macro-enabled document)
			document.SaveToFile("Sample.docm", FileFormat.Docm)

			' Dispose of the document object when finished using it
			document.Dispose()

			'Launching the MS Word file.
			WordDocViewer("Sample.docm")
		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace

File: CS-Examples/30_Macros/Macros.cs
using System;
using System.Drawing;
using System.IO;
using System.Windows.Forms;
using System.Xml.XPath;
using Spire.Doc;
using Spire.Doc.Documents;
using Spire.Doc.Fields;

namespace Macros
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new instance of Document
            Document document = new Document();

            // Load the Word document from the specified file that may contain VBA macros
            document.LoadFromFile(@"../../../../../../Data/Macros.docm", FileFormat.Docm);

            // Save the document to a new file with the specified name and format (Docm for macro-enabled document)
            document.SaveToFile("Sample.docm", FileFormat.Docm);

            // Dispose of the document object when finished using it
            document.Dispose();

            //Launching the Word file.
            WordDocViewer("Sample.docm");
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}
